### PR TITLE
Added Perfect Number

### LINKED ISSUE #1159: Perfect number in c , c++ , python , go and java
Perfect number is a positive integer that is equal to the sum of its proper divisors.

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #1159

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Implementation for Perfect Number in C, C++, Java, Python, Golang

#### Languages Used:
- C
- C++
- Java
- Python
- Golang

#### Files Added:
- Perfect_Number.c 
- Perfect_Number.cpp
- Perfect_Number.java
- Perfect_Number.py
- Perfect_Number.go

### CODE REVIEW

**Comment by MastersAbh** [comment]
Hi,
You must wait for the issue to get assigned to you.

File: Perfect_Number/Perfect_Number.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Perfect number is a positive integer that is equal to the sum of its proper divisors.
+
+package main
+
+import (
+	"fmt"
```
Comment (jainaman224) [inline Perfect_Number/Perfect_Number.go]: Indentation is incorrect.

File: Perfect_Number/Perfect_Number.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Perfect number is a positive integer that is equal to the sum of its proper divisors.
+
+public class Perfect
+{
+    public static void main(String[] args) 
+    {
+        int num = 28, sum = 0;
+        
+        // Calculating the sum of divisors
+        for(int i = 1; i < num; i++)
```
Comment (jainaman224) [inline Perfect_Number/Perfect_Number.java]: Space after `for`, `if`, `while` etc looks good.

**Comment by somya-kapoor** [comment]
@jainaman224 Done. Please review.

File: Perfect_Number/Perfect_Number.c [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// Perfect number is a positive integer that is equal to the sum of its proper divisors.
+
+#include <stdio.h>
+ 
+int main()
+{
+    int num, sum = 0, i;
+
```
Comment (MastersAbh) [inline Perfect_Number/Perfect_Number.c]: Please remove this unnecessary blank line.

File: Perfect_Number/Perfect_Number.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// Perfect number is a positive integer that is equal to the sum of its proper divisors.
+
+#include <iostream>
+using namespace std;
+ 
+int main()
+{
+    int num, rem, sum = 0, i;
+
```
Comment (MastersAbh) [inline Perfect_Number/Perfect_Number.cpp]: Please remove this blank line

File: Perfect_Number/Perfect_Number.cpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    if (sum == num)
+        cout << "It is a perfect number";
+    else
+        cout << "It is not a perfect number";
+    return 0;
+}
+
+/*
+Enter a Number: 28
+It is a perfect number*/
```
Comment (MastersAbh) [inline Perfect_Number/Perfect_Number.cpp]: A space before */

File: Perfect_Number/Perfect_Number.c [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    if (sum == num)
+        printf("It is a perfect number");
+    else
+        printf("It is not a perfect number");
+    return 0;
+}
+
+/*
+Enter a Number: 28
+It is a perfect number*/
```
Comment (MastersAbh) [inline Perfect_Number/Perfect_Number.c]: A space before */

**Comment by somya-kapoor** [comment]
@MastersAbh Done. Please review.

**Pull request merged**